### ISSUE TITLE
Japanese Texts are smaller than alphabet Texts even though they are same FontSize

### ISSUE THREAD

**Issue by najeira** [flutter/flutter#40209]
## Steps to Reproduce

![Screenshot_20190911-123128](https://user-images.githubusercontent.com/1098127/64666344-ec848e00-d490-11e9-97ce-e321d669749d.jpg)

This screenshot was recorded on Galaxy S7 Edge.

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Text heights',
      home: HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  @override
  HomePageState createState() => HomePageState();
}

class HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Text heights"),
      ),
      body: DefaultTextStyle(
        style: TextStyle(fontSize: 20.0),
        child: Column(
          children: <Widget>[
            Container(
              padding: const EdgeInsets.all(20.0),
              child: Row(
                children: <Widget>[
                  Container(
                    color: Colors.red,
                    child: Text(
                      "ABC",
                    ),
                  ),
                  Container(
                    color: Colors.green,
                    child: Text(
                      "あいう",
                    ),
                  ),
                ],
              ),
            ),
            Container(
              padding: const EdgeInsets.all(20.0),
              child: Row(
                children: <Widget>[
                  Container(
                    color: Colors.red,
                    child: Text(
                      "ABC",
                    ),
                  ),
                  Container(
                    color: Colors.blue,
                    child: Text(
                      "😀😀😀",
                    ),
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

## Logs

```
[✓] Flutter (Channel stable, v1.9.1+hotfix.2, on Mac OS X 10.14.4 18E226, locale ja-JP)
    • Flutter version 1.9.1+hotfix.2 at /Applications/flutter
    • Framework revision 2d2a1ffec9 (4 days ago), 2019-09-06 18:39:49 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0


[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/najeira/Library/Android/sdk
    • Android NDK at /Users/najeira/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/najeira/Library/Android/sdk
    • ANDROID_SDK_ROOT = /Users/najeira/Library/Android/sdk
    • Java binary at: /Users/najeira/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/191.5791312/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 10.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.2, Build version 10E125
    • CocoaPods version 1.7.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] Android Studio (version 3.5)
    • Android Studio at /Users/najeira/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/191.5791312/Android Studio.app/Contents
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] IntelliJ IDEA Ultimate Edition (version 2018.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 31.3.1
    • Dart plugin version 181.4203.498

[✓] VS Code (version 1.38.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.4.1
```

## Comment

Until 1.7.1, Japanese Texts are taller than alphabet Texts.

See:

* https://github.com/flutter/flutter/issues/18473
* https://medium.com/@najeira/control-text-height-using-strutstyle-4b9b5151668b

**Comment by GaryQian** [comment]
https://github.com/flutter/engine/pull/9041 Changed `height` property behavior. If there is an instance of `height` being set, this may be causing this.

Beyond that, I think we have been seeing some font loading changes as a result of iOS 13.

**Comment by najeira** [comment]
This issue has not been resolved since flutter/engine#9041. I confirmed that it also occurred with current beta 1.5.17.

It is not practical to set height all Text widgets. It should work fine by default.

This problem does not occur with native text drawing on Android or iOS.

Here's the screen shot when height is set to 1.2:

![Screenshot_20200327-105938](https://user-images.githubusercontent.com/1098127/77713804-afda4780-701a-11ea-90ef-dc04ef74aef5.jpg)

The Japanese position appears to be shifted down.
